src/Sylius/Component/Core/Updater/UnpaidOrdersStateUpdater.php

```diff
@@ -29,24 +30,36 @@ final class UnpaidOrdersStateUpdater implements UnpaidOrdersStateUpdaterInterfac
     /** @var string */
     private $expirationPeriod;
 
+    /** @var LoggerInterface */
+    private $logger;
+
     /**
      * @param string $expirationPeriod
      */
     public function __construct(
         OrderRepositoryInterface $orderRepository,
         Factory $stateMachineFactory,
-        $expirationPeriod
+        $expirationPeriod,
+        LoggerInterface $logger
```

Comment:
```suggestion
        LoggerInterface $logger = null
```

You need to allow this to be null, else it's a BC break and can't be in 1.x. You could (should) trigger a deprecation warning if this is null

Reply:
Done